processor/filterprocessor/filter_processor_traces.go

```diff
@@ -0,0 +1,127 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterspan"
+)
+
+type filterSpanProcessor struct {
+	cfg     *Config
+	include filterspan.Matcher
+	exclude filterspan.Matcher
+	logger  *zap.Logger
+}
+
+func newFilterSpansProcessor(logger *zap.Logger, cfg *Config) (*filterSpanProcessor, error) {
+	if cfg.Spans.Include == nil && cfg.Spans.Exclude == nil {
+		return nil, nil
+	}
+
+	inc, exc, err := createSpanMatcher(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	includeMatchType := ""
```

Comment:
If you are using this only in the logs, perhaps `<none>` would be a better default?